core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -29,6 +30,16 @@ import org.apache.spark.serializer.{SerializerInstance, Serializer}
 import org.apache.spark.util.{CompletionIterator, Utils}
 
 /**
+ * Factory class that creates iterators to read records from fetched blocks
+ */
+trait BlockRecordIteratorFactory {
```

Comment:
This was meant to be a developer API . I can update this to be a Java interface.